docs/contributing/authoring-polyfills.html

```diff
@@ -0,0 +1,173 @@
+{{>header}}
+
+{{>nav this}}
+
+	<div class="o-techdocs-main o-techdocs-content">
+
+		<h1>Authoring polyfills</h1>
+
+		<p>Polyfills are located in the <code>/polyfills</code> directory and organised by feature name.</p>
+
+		<h2 id='files'>Files</h2>
+
+		<p>Each polyfill directory (and subdirectory) is structured like this:</p>
+
+		<ul>
+			<li><code>polyfill.js</code>: Required. Code to apply the polyfill behaviour.  See <a href="#configuration">configuration</a>.</li>
+			<li><code>detect.js</code>: A single expression that returns true if the feature is present in the browser (and the polyfill is therefore not required), false otherwise.  Can be an IIFE.  If not present, polyfill cannot be gated (wrapped in a feature-detect).  <strong>Must not have a trailing semicolon</strong>.</li>
+			<li><code>tests.js</code>: A set of tests written using <a href="http://mochajs.org/">mocha</a> and <a href="https://github.com/LearnBoost/expect.js/">expect</a>, to test the feature.  See <a href="#test-authoring-guidelines">test authoring guidelines</a>.</li>
+			<li><code>update.task.js</code>: Optional alternative to <code>polyfill.js</code>. This file is used in combination with the Grunt task <code>updatelibrary</code> to create a local copy of an external polyfill.</li>
```

Comment:
Lots of mandatory requirements reduces demand to participate, so we have to strike a balance.